src/internal/connector/exchange/service_iterators.go

```diff
@@ -105,7 +102,36 @@ func FilterContainersAndFillCollections(
 
 		// add the current path for the container ID to be used in the next backup
 		// as the "previous path", for reference in case of a rename or relocation.
-		currPaths[cID] = dirPath.Folder()
+		currPaths[cID] = currPath.String()
+	}
+
+	// any path that wasn't present in the resolver was deleted by the user.
+	// relocations and renames will have removed the dir by id earlier.  What's
+	// left in deletedPaths are only the previous paths that did not appear as
+	// children of the root.
+	for fID, ps := range deletedPaths {
+		service, err := createService(qp.Credentials)
+		if err != nil {
+			errs = support.WrapAndAppend(qp.ResourceOwner, err, errs)
+			continue
+		}
+
+		prevPath, err := pathFromPrevString(ps)
+		if err != nil {
+			logger.Ctx(ctx).Error(err)
+			continue
+		}
+
+		edc := NewCollection(
```

Comment:
you could probably use an empty `MetadataCollection` here or something. I know it's kind of jank, but I'm also a bit uncomfortable with having to make a service for a collection that has no items and really won't have `Items()` called on it at all

Reply:
Not sure I follow your meaning of `a service` in this case.

Reply:
ah, sorry for being unclear. The ExchangeCollection requires a `graph.Servicer` instance because it assumes it will need to fetch items. However, Collections that represent deleted folders really just need to have the state set and implement the proper interface. This means it's kind of useless to create `service` on line 145 and use it in this call